src/main/java/com/researchspace/webapp/integrations/fieldmark/FieldmarkApiController.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.client.HttpServerErrorException;
+
+@Slf4j
+@ApiController
+public class FieldmarkApiController extends BaseApiController implements FieldmarkApi {
+
+  private @Autowired SampleTemplatesApiController sampleTemplatesApiController;
+  private @Autowired InventoryFilesApiController inventoryFilesApiController;
+  private @Autowired SamplesApiController samplesApiController;
+  private @Autowired ContainersApiController containersApiController;
+
+  private @Autowired FieldmarkServiceClientAdapter fieldmarkServiceClientAdapter;
+
+  @Override
+  public List<FieldmarkNotebook> getNotebooks(@RequestAttribute(name = "user") User user)
+      throws MalformedURLException, URISyntaxException, BindException {
+    try {
+      return fieldmarkServiceClientAdapter.getFieldmarkNotebookList(user);
+    } catch (HttpServerErrorException serverEx) {
+      log.error(
+          "The list of notebooks cannot be fetched because of an error on the Fieldmark server: "
+              + serverEx.getMessage());
+      BindingResult errors = new BeanPropertyBindingResult(null, "fieldmarkServer");
+      errors.rejectValue(
+          "",
+          "errors.fieldmark.import",
+          new Object[] {},
+          "Error fetching notebooks due to the Fieldmark server");
+      throwBindExceptionIfErrors(errors);
+    }
+    return null;
+  }
+
+  @Override
+  public FieldmarkApiResult importNotebook(
+      @PathVariable(name = "id") String notebookId, @RequestAttribute(name = "user") User user)
+      throws BindException, IOException {
+
+    FieldmarkNotebookDTO fieldmarkNotebookDTO = null;
+    BindingResult errors = null;
+    try {
+      fieldmarkNotebookDTO = fieldmarkServiceClientAdapter.getFieldmarkNotebook(user, notebookId);
+    } catch (IOException e) {
+      log.error(
+          "The notebookID \"\" has not being imported from Fieldmark "
+              + "cause to the following error:"
+              + e.getMessage());
+      errors = new BeanPropertyBindingResult(fieldmarkNotebookDTO, "notebookId");
+      errors.rejectValue(
+          "",
+          "errors.fieldmark.import",
+          new Object[] {},
+          "Error importing notebook \"" + notebookId + "\" from fieldmark");
+
+    } catch (HttpServerErrorException serverEx) {
+      log.error(
+          "The notebook cannot be fetched because of an error on the Fieldmark server: "
+              + serverEx.getMessage());
+      errors = new BeanPropertyBindingResult(null, "fieldmarkServer");
+      errors.rejectValue(
+          "",
+          "errors.fieldmark.import",
+          new Object[] {},
+          "Error importing notebook \""
+              + notebookId
+              + "\" due to Fieldmark server "
+              + "unavailable");
+    }
```

Comment:
That should be correct, becasue potentially iut does not have to tell you that the notebookId is wrong, otherwise an attacker could use the response to guess the notebookId. 
So, if you put a wrong notebookId, he does not distinguish if it is wrong or it is the notebookId of another user that you cannot access. Does it make sense?

Reply:
hmm, that makes sense to a degree, but we shouldn't be returning 500 internal server error in that case. If the 3rd party API returns 401 unauthorized then our API should be returning 403 forbidden I think. The `catch` block above handles the  `HttpServerErrorException`, but not `HttpClientErrorException`, we should be catching both.

Reply:
need to catch `HttpClientErrorException` as well though

Reply:
👷‍♂️

Reply:
done

Reply:
so now the following JSON will be returned when you put a notebookId that is not owned by the user:
```
{
  "status": "BAD_REQUEST",
  "httpCode": 400,
  "internalCode": 40002,
  "message": "Errors detected : 1",
  "messageCode": null,
  "errors": [
    "fieldmarkApiImportRequest: Error importing notebook \"1726126204618-rspace-igsn-demo__\" due to the following error: 401 Unauthorized: [no body]"
  ],
  "iso8601Timestamp": "2024-11-28T15:27:45.884006Z",
  "data": {
    "validationErrors": [
      {
        "field": null,
        "rejectedValue": null,
        "objectName": "fieldmarkApiImportRequest",
        "message": "Error importing notebook \"1726126204618-rspace-igsn-demo__\" due to the following error: 401 Unauthorized: [no body]"
      }
    ]
  }
}
```